container-images/build_llama_and_whisper.sh

```diff
@@ -0,0 +1,113 @@
+  local intel_rpms=("intel-oneapi-mkl-sycl-devel" "intel-oneapi-dnnl-devel" "intel-oneapi-compiler-dpcpp-cpp" "intel-level-zero" "oneapi-level-zero" "oneapi-level-zero-devel" "intel-compute-runtime")
+
+  LLAMA_CPP_CMAKE_FLAGS+=("-DGGML_CCACHE=OFF" "-DGGML_NATIVE=OFF" "-DBUILD_SHARED_LIBS=NO" "-DLLAMA_CURL=ON")
+  WHISPER_CPP_CMAKE_FLAGS+=("-DGGML_CCACHE=OFF" "-DGGML_NATIVE=OFF" "-DBUILD_SHARED_LIBS=NO")
+
+  case ${container_image} in
+    ramalama)
+      dnf --enablerepo=ubi-9-appstream-rpms install -y "${common_rpms[@]}"
+      dnfPrepUbi
+      dnf --enablerepo=ubi-9-appstream-rpms install -y mesa-vulkan-drivers "${vulkan_rpms[@]}"
+      LLAMA_CPP_CMAKE_FLAGS+=("-DGGML_KOMPUTE=ON" "-DKOMPUTE_OPT_DISABLE_VULKAN_VERSION_CHECK=ON")
+    ;;
+    rocm)
+      dnf --enablerepo=ubi-9-appstream-rpms install -y "${common_rpms[@]}"
+      dnfPrepUbi
+      dnf install -y "${vulkan_rpms[@]}" rocm-dev hipblas-devel rocblas-devel
+      LLAMA_CPP_CMAKE_FLAGS+=("-DGGML_HIP=ON" "-DAMDGPU_TARGETS=${AMDGPU_TARGETS:-gfx1010,gfx1030,gfx1032,gfx1100,gfx1101,gfx1102}")
+      WHISPER_CPP_CMAKE_FLAGS+=("-DGGML_HIP=ON" "-DAMDGPU_TARGETS=${AMDGPU_TARGETS:-gfx1010,gfx1030,gfx1032,gfx1100,gfx1101,gfx1102}")
+    ;;
+    cuda)
+      dnf install -y "${common_rpms[@]}" gcc-toolset-12
+      . /opt/rh/gcc-toolset-12/enable
+      LLAMA_CPP_CMAKE_FLAGS+=("-DGGML_CUDA=ON" "-DCMAKE_EXE_LINKER_FLAGS=-Wl,--allow-shlib-undefined")
+      WHISPER_CPP_CMAKE_FLAGS+=("-DGGML_CUDA=ON" "-DCMAKE_EXE_LINKER_FLAGS=-Wl,--allow-shlib-undefined")
+      install_prefix=/llama-cpp
+    ;;
+    vulkan)
+      dnf --enablerepo=ubi-9-appstream-rpms install -y "${common_rpms[@]}"
+      dnfPrepUbi
+      dnf --enablerepo=ubi-9-appstream-rpms install -y mesa-vulkan-drivers "${vulkan_rpms[@]}"
+      LLAMA_CPP_CMAKE_FLAGS+=("-DGGML_VULKAN=1")
+      WHISPER_CPP_CMAKE_FLAGS+=("-DGGML_VULKAN=1")
+    ;;
+    asahi)
+      dnf copr enable -y @asahi/fedora-remix-branding
+      dnf install -y asahi-repos
+      dnf install -y mesa-vulkan-drivers "${vulkan_rpms[@]}" "${common_rpms[@]}"
+      LLAMA_CPP_CMAKE_FLAGS+=("-DGGML_VULKAN=1")
+      WHISPER_CPP_CMAKE_FLAGS+=("-DGGML_VULKAN=1")
+    ;;
+    intel-gpu)
+      dnf install -y ${common_rpms[@]} ${intel_rpms[@]}
+      LLAMA_CPP_CMAKE_FLAGS+=("-DGGML_SYCL=ON" "-DCMAKE_C_COMPILER=icx" "-DCMAKE_CXX_COMPILER=icpx")
+      WHISPER_CPP_CMAKE_FLAGS+=("-DGGML_SYCL=ON" "-DCMAKE_C_COMPILER=icx" "-DCMAKE_CXX_COMPILER=icpx")
+      install_prefix=/llama-cpp
+      source /opt/intel/oneapi/setvars.sh
+    ;;
+  esac
+
+  CMAKE_FLAGS+=("${WHISPER_CPP_CMAKE_FLAGS[@]}")
+  cloneAndBuild https://github.com/ggerganov/whisper.cpp ${whisper_cpp_sha} ${install_prefix}
+  CMAKE_FLAGS+=(${LLAMA_CPP_CMAKE_FLAGS[@]})
+  cloneAndBuild https://github.com/ggerganov/llama.cpp ${llama_cpp_sha} ${install_prefix}
+  dnf -y clean all
+  rm -rf /var/cache/*dnf* /opt/rocm-*/lib/*/library/*gfx9*
```

Comment:
This was where the bulk of the trimming was done for rocm in the past. Not sure why it's ballooned again.

I know this sounds a little hard, but I'd revert this shell script back to it's original state and introduce just enough changes to get "intel-gpu" working with this script. There's no need to rewrite the whole script, it's asking for breakages.

Reply:
Understood.  It was a wild hair 🤓